EIPS/eip-6147.md

```diff
@@ -0,0 +1,272 @@
+    }
+    
+    /// @notice Check the guard address
+    /// @dev The zero address indicates there is no guard
+    /// Throws if `tokenId` is not valid NFT
+    /// @param tokenId The NFT to check the guard address for
+    /// @return The guard address
+    function checkGuard(uint256 tokenId) internal view returns (address) {
+        address guard = guardOf(tokenId);
+        address sender = _msgSender();
+        if (guard != address(0)) {
+            require(guard == sender, "sender is not guard of token");
+            return guard;
+        }else{
+            return address(0);
+        }
+    }
+
+    ///@dev When burning, delete `token_guard_map[tokenId]`
+    function _burn(uint256 tokenId) internal virtual override {
+        address guard=guardOf(tokenId);
+        super._burn(tokenId);
+        delete token_guard_map[tokenId];
+        emit UpdateGuardLog(tokenId, address(0), guard);
+    }
+ 
+    /// @dev Before transferring the NFT, need to check the gurard address
+    function transferFrom(address from,address to,uint256 tokenId) public virtual override {
+        address guard;
+        address new_from = from;
+        if (from != address(0)) {
+            guard = checkGuard(tokenId);
+            new_from = ownerOf(tokenId);
+        }
+        if (guard == address(0)) {
+            require(
+                _isApprovedOrOwner(_msgSender(), tokenId),
+                "ERC721: transfer caller is not owner nor approved"
+            );
+        }
+        _transfer(new_from, to, tokenId);
+    }
+
+    /// @dev Before safe transferring the NFT, need to check the gurard address
+    function safeTransferFrom(address from,address to,uint256 tokenId,bytes memory _data) public virtual override {
+        address guard;
+        address new_from = from;
+        if (from != address(0)) {
+            guard = checkGuard(tokenId);
+            new_from = ownerOf(tokenId);
+        }
+        if (guard == address(0)) {
+            require(
+                _isApprovedOrOwner(_msgSender(), tokenId),
+                "ERC721: transfer caller is not owner nor approved"
+            );
+        }
+        _safeTransfer(from, to, tokenId, _data);
+    }
+
+    /// @dev See {IERC165-supportsInterface}.
+    function supportsInterface(bytes4 interfaceId) public view virtual override returns (bool) {
+        return interfaceId == type(IERC721QS).interfaceId || super.supportsInterface(interfaceId);
+    }
+}
+
+```
+
+
+## Security Considerations
+
+When an NFT has a `guard`, even if an address is authorized as an operator through `approve` or `setApprovalForAll`, the operator still has no right to transfer the NFT.
+
+For NFT trading platforms that trade through `setApprovalForAll` + holder's signature, when NFT has `guard`, it cannot be traded. It is recommended to prevent such pending orders by checking the interface beforehand.
+
+## Copyright
+
+Copyright and related rights waived via CC0.
```

Comment:
Please use the exact text from the [EIP template](https://github.com/ethereum/EIPs/blob/master/eip-template.md#copyright)